mi/instrument/seabird/sbe16plus_v2/dosta/driver.py

```diff
@@ -0,0 +1,294 @@
+class InstrumentDriver(SingleConnectionInstrumentDriver):
+
+    def _build_protocol(self):
+        """
+        Construct the driver protocol state machine.
+        """
+        self._protocol = Protocol(Prompt, NEWLINE, self._driver_event)
+
+
+####################################################################################
+# Command Protocols - read-only, the attached CTD is used to control the instrument
+####################################################################################
+class Protocol(CommandResponseInstrumentProtocol):
+    """
+    Instrument protocol class for SBE16 DOSTA driver.
+    """
+
+    particles = [
+        DoSampleParticle,
+    ]
+
+    def __init__(self, prompts, newline, driver_event):
+        """
+        @param prompts A BaseEnum class containing instrument prompts.
+        @param newline The SBE16 newline.
+        @param driver_event Driver process event callback.
+        """
+        CommandResponseInstrumentProtocol.__init__(self, prompts, newline, driver_event)
+
+        # This driver does not process commands, the finite state machine and handlers are stubs
+        self._protocol_fsm = ThreadSafeFSM(ProtocolState, ProtocolEvent,
+                                           ProtocolEvent.ENTER, ProtocolEvent.EXIT)
+
+        handlers = {
+            ProtocolState.UNKNOWN: {
+                (ProtocolEvent.ENTER, self._handler_state_change()),
+                (ProtocolEvent.EXIT, self._handler_pass_through()),
+                (ProtocolEvent.DISCOVER, self._handler_unknown_discover()),
+            },
+            ProtocolState.COMMAND: {
+                (ProtocolEvent.ENTER, self._handler_state_change()),
+                (ProtocolEvent.EXIT, self._handler_pass_through()),
+                (ProtocolEvent.GET, self._handler_pass_through()),
+            },
+        }
+
+        for state in handlers:
+            for event, handler in handlers[state]:
+                self._protocol_fsm.add_handler(state, event, handler)
+
+        self._build_param_dict()
+        self._build_command_dict()
+        self._build_driver_dict()
+
+        self._protocol_fsm.start(ProtocolState.UNKNOWN)
+
+        self._chunker = StringChunker(self.sieve_function)
+
+    @staticmethod
+    def sieve_function(raw_data):
+        """ The method that splits samples
+        Over-ride sieve function to handle additional particles.
+        """
+        matchers = []
+        return_list = []
+
+        matchers.append(DoSampleParticle.regex_compiled())
+        for matcher in matchers:
+            for match in matcher.finditer(raw_data):
+                return_list.append((match.start(), match.end()))
+
+        return return_list
+
+    def _build_command_dict(self):
+        self._cmd_dict.add(Capability.DISCOVER, display_name='Discover', timeout=1)
+
+    def _build_param_dict(self):
+        pass
+        # self._param_dict.add(Parameter.OPTODE,
```

Comment:
Do these comments need to remain?

Reply:
I left them in because someone in the future may want to add configuration parameters for the DOSTA read-only driver (it is controlled by the CTD driver, so you can't change anything, but it may be useful to know that these parameters were set).